pkg/apis/pipeline/v1/container_validation.go

```diff
@@ -18,45 +18,79 @@ package v1
 
 import (
 	"context"
+	"fmt"
 	"strings"
 
+	"net/url"
+
 	"github.com/tektoncd/pipeline/pkg/apis/config"
 	"k8s.io/apimachinery/pkg/util/validation"
 	"knative.dev/pkg/apis"
 )
 
-// Validate ensures that a supplied Ref field is populated
-// correctly. No errors are returned for a nil Ref.
-func (ref *Ref) Validate(ctx context.Context) (errs *apis.FieldError) {
-	if ref == nil {
-		return errs
-	}
-
+func validateRef(ctx context.Context, refName string, refResolver ResolverName, refParams Params) (errs *apis.FieldError) {
```

Comment:
nit: `validateRef` looks more likely to be in `taskref_validation` than `container_validation`

Reply:
This is for validating the StepAction Ref. I think that's in container validation. 
There is a separate `validateTaskRef` which is in taskref validation.

Reply:
I am not sure I understand - ain't this change here using `validateRef` as a function that's used commonly at container_validation, taskref_validation and pipeline_validation?

And maybe as you pointed explicitly name it as `validateStepActionRef` might untangle this?

Reply:
It's tied to validating the ref. The field name for  StepAction Ref is `ref`. That's why I went with `validateRef` since it is validating the `ref` field.